contrib/world_builder/include/world_builder/features/continental_plate.h

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright (C) 2018 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef _world_feature_features_continental_plate_h
+#define _world_feature_features_continental_plate_h
+
+#include <world_builder/features/interface.h>
+#include <world_builder/world.h>
+
+
+namespace WorldBuilder
+{
+  namespace Features
+  {
+
+    class ContinentalPlate : public Interface
+    {
+      public:
+        /**
+         * constructor
+         */
+        ContinentalPlate(WorldBuilder::World *world);
+
+        /**
+         * Destructor
+         */
+        ~ContinentalPlate();
+
+        /**
+         * Read in the world builder file
+         */
+        virtual
+        void read(ptree &property_tree);
+
+        /**
+         * Returns a temperature based on the given position
+         */
+        virtual
+        double temperature(const Point<3> position,
+                           const double depth,
+                           const double gravity,
+                           double temperature) const;
+
+        /**
+         * Returns a value for the reqeusted composition (0 is not present,
+         * 1 is present) based on the given position and
+         */
+        virtual
```

Comment:
you could change this to return a bool instead and convert it to double inside the plugin. Maybe that is a more natural interface?

Reply:
Yes, that makes a lot of sence